oauthproxy.go

```diff
@@ -851,13 +861,13 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
 
 	case ErrNeedsLogin:
 		// we need to send the user to a login screen
-		if isAjax(req) {
+		if isAjax(req) && !p.APIMode {
 			// no point redirecting an AJAX request
 			p.ErrorJSON(rw, http.StatusUnauthorized)
```

Comment:
If we are extending the JSON payload to have more context on the error, I think we should add that message here potentially to align everything?